crates/shared/src/order_validation.rs

```diff
@@ -379,15 +379,23 @@ impl OrderValidating for OrderValidator {
             return Err(ValidationError::ZeroAmount);
         }
 
-        let liquidity_owner = self.liquidity_order_owners.contains(&owner);
+        let class = if self.liquidity_order_owners.contains(&owner) {
+            OrderClass::Liquidity
+        } else if self.enable_limit_orders && order.data.fee_amount.is_zero() {
+            OrderClass::Limit
```

Comment:
Couldn't we just return `Liquidity` if `!self.enable_limit_orders && order.data.fee_amount.is_zero()`?
Then we wouldn't have to reclassify this particular order and the code becomes easier to understand. Or am I missing some side effect we need?
Also wouldn't we still allow market orders with 0 fee if they are within market price with this PR?

Reply:
The reclassification is needed for market orders that are out of price. We don't have that information at this point in the code.